content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,151 @@
+
+### Table of contents
+
+1. [Introduction](#introduction)
+2. [Installing dependencies](#installing-dependencies)
+3. [Downloading the pre-trained translation model](#downloading-the-pre-trained-translation-model)
+4. [Building the web app for interaction using Gradio](#building-the-web-app-for-interaction-using-gradio)
+5. [Wrapping up](#wrapping-up)
+
+### Introduction
+
+[Hugging Face](https://huggingface.co/transformers/) is a great open-source library that is doing powerful work in the Natural Language Processing (NLP) space. The library has a bunch of pre-trained models which you can leverage or fine-tune. Rather than going ahead and training a huge language model, one can leverage this library instead.     
+
+[Gradio](https://gradio.app/), on the other hand, allows us to quickly create customizable UI components around PyTorch models.
+
+### Installing dependencies
+
+Our main dependencies are going to be PyTorch, Hugging Face Transformers, and Gradio. Let's go ahead and install them.
+
+Let's first install PyTorch. This is available [here](https://pytorch.org/) on PyTorch's website. On the website, find the `Install` button and press it. On the install page, you will see a walkthrough section similar to the one shown below:
+
+![Installing PyTorch](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/installing-pytorch.png)
+
+*[Image Source: PyTorch](https://pytorch.org/)*
+
+The commands for installing PyTorch vary depending on factors such as your OS, package of choice (pip, conda, source), language, and compute platform (CUDA, CPU, or ROCm). So, make sure you select your preference. It doesn't have to look like mine. Mine's shown below:
+
+![Preference](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/pytorch-preference.png)
+
+*[Image Source: PyTorch](https://pytorch.org/)*
+
+To install PyTorch on my machine, I've selected the Long Term Support (LTS 1.8.1) build, Linux OS, `pip` package, Python programming language, and CPU as my compute platform of choice. If your machine has GPU support, choose accordingly.  
+
+After selecting those options, this is the command you're going to run to install PyTorch:
+
+```python
+    pip3 install torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio==0.8.1 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
+```
+> An important point to note is that recently, Google Colab comes pre-installed with PyTorch. You may need to ignore the PyTorch installation if you're using Colab. Only follow the steps if installing on your machine.
+
+Let's now go ahead and install our other dependencies, the Hugging Face Transformers and Gradio.
+
+```python
+    pip install transformers ipywidgets gradio --upgrade
+```
```

Comment:
This could be put in a bash codeblock as well. Also, the actual command need not be indented.